### PR TITLE
[Wasm] Add AOT CI Leg

### PR DESCRIPTION
**Pull request by steveisok**
Contributes to https://github.com/dotnet/runtime/issues/46633

With the `EnableAggressiveTrimming` flag added in https://github.com/dotnet/runtime/pull/46367, the library testing size has been reduced to a much smaller size (~2MB), making AOT compilation of these library tests more reasonable. This PR stands-up a lane on CI to see how far wasm AOT library tests can go.

A significant number of library tests have failed on CI. Most of them have failed because they do not include `Microsoft.DotNet.RemoteExecutor`, which had been added to https://github.com/dotnet/runtime/blob/master/eng/testing/ILLink.Descriptor.xunit.xml due to a linker bug https://github.com/mono/linker/issues/1713.

The [failures](https://github.com/dotnet/runtime/files/5796961/initial_wasm_AOT_lane_standup.txt) in the lane have been parsed out to the following [test suites](https://github.com/dotnet/runtime/files/5796970/test_suites_missing_Microsoft_DotNet_RemoteExecutor.txt) that are missing `Microsoft.DotNet.RemoteExecutor`.

The test suites above have been re-ran locally with the exclusion of `Microsoft.DotNet.RemoteExecutor` to verify whether the test fails or passes, and suites continuing to fail have been added to project exclusions in `tests.proj`. A significant number of test suites fall under several categories of failures, but for simplicity, they are listed under just one, and issues have been created for each unique failure.

In lieu of running all the library tests suites in AOT mode on CI (estimated to be ~36 hrs = 240 test suites * ~9min AOT time per test suite), only System.Buffers.Tests.csproj will be ran in AOT mode to standup the leg. In the meantime, alternative solutions are being worked on.

The CI lane will be located in `runtime-staging.yml` to avoid blocking and failing builds.

### CODE REVIEW

File: eng/pipelines/runtime.yml [hunk 1]
```diff
@@ -271,6 +271,47 @@ jobs:
           eq(variables['monoContainsChange'], true),
           eq(variables['isFullMatrix'], true))
 
+#
+# Build the whole product using Mono and run libraries tests, multi-scenario
+#
+- template: /eng/pipelines/common/platform-matrix.yml
```
Comment (safern) [inline eng/pipelines/runtime.yml:277]: I would suggest adding this to runtime-staging once this is ready to be merged (I know we're on early process gathering info on how many tests fail, etc) to get it stable.

Comment (mdh1418) [reply eng/pipelines/runtime.yml]: The lane passed after running just `System.Buffers.Tests.csproj` :) 
https://dev.azure.com/dnceng/public/_build/results?buildId=948229&view=logs&j=f217f715-90ad-5404-53bd-b93e3b80f465&t=9df76bd4-3041-5537-002b-9e29a53263b9
Moved the block to `runtime-staging.yml`

**Review by akoeplinger — APPROVED** [review]

File: src/libraries/tests.proj [hunk 2]
```diff
@@ -159,6 +159,104 @@
...
+
+    <!-- https://github.com/dotnet/runtime/issues/46768 -->
+    <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Diagnostics.DiagnosticSource/tests/TestWithConfigSwitches/System.Diagnostics.DiagnosticSource.Switches.Tests.csproj" />
+
+    <!-- https://github.com/dotnet/runtime/issues/46856 -->
+    <ProjectExclusions Include="$(MSBuildThisFileDirectory)/System.Diagnostics.FileVersionInfo/tests/System.Diagnostics.FileVersionInfo.Tests/System.Diagnostics.FileVersionInfo.Tests.csproj" />
+
+    <!-- Run only a small randomly chosen set of passing test suites on CI because running all suites is expected to take longer than a day -->
+    <ProjectExclusions Include="$(MSBuildThisFileDirectory)*\tests\**\*.Tests.csproj" />
+    <ProjectExclusions Remove="$(MSBuildThisFileDirectory)System.Buffer.Tests\tests\System.Buffer.Tests.csproj" />
```
Comment (MichalStrehovsky) [inline src/libraries/tests.proj:257]: @steveisok Btw, the $(MSBuildThisFileDirectory)System.Buffer.Tests directory doesn't exist, so this is not testing anything.

```suggestion
<ProjectExclusions Remove="$(MSBuildThisFileDirectory)System.Buffer\tests\System.Buffer.Tests.csproj" />
```

Comment (mdh1418) [reply src/libraries/tests.proj:257]: Thanks for the catch! There is more work being done in https://github.com/dotnet/runtime/pull/48226 that removes this line, and later the exclusions will be revisited and updated according to this tracking issue https://github.com/dotnet/runtime/issues/48079.

**Pull request merged**